File: airport-management-backend/src/main/java/com/airportmanagementbackend/service/CrewMemberService.java
package com.airportmanagementbackend.service;

import com.airportmanagementbackend.common.dto.crewmember.CrewMemberDto;
import java.util.List;
import java.util.Optional;

public interface CrewMemberService {

  List<CrewMemberDto> findAll();

  List<CrewMemberDto> findAllByFlightId(Long flightId);

  Optional<CrewMemberDto> findById(Long id);

  CrewMemberDto create(CrewMemberDto toCreate);

  CrewMemberDto update(CrewMemberDto toUpdate);

  boolean delete(Long id);

}

File: airport-management-frontend/src/app/core/services/crew.member.service.ts
import {Injectable} from "@angular/core";
import {BACKEND_APIS} from "../util/backend-apis";
import {HttpClient} from "@angular/common/http";
import {Observable} from "rxjs";
import {CrewMember} from "../../shared/models/CrewMember";
import {CrewMembersWithFlights} from "../../shared/models/CrewMembersWithFlights";

@Injectable({
  providedIn: "root"
})
export class CrewMemberService {
  private baseUrl = BACKEND_APIS.crewMemberApi

  constructor(private httpClient: HttpClient) {
  }

  public getCrewMembers(): Observable<{crewMembers: CrewMember[]}> {
    return this.httpClient.get<{crewMembers: CrewMember[]}>(`${this.baseUrl}/list`);
  }

  public getCrewMemberById(id: number): Observable<CrewMembersWithFlights> {
    return this.httpClient.get<CrewMembersWithFlights>(`${this.baseUrl}/get/by-id?id=${id}`)
  }

  public createCrewMember(toCreate: CrewMember): Observable<CrewMembersWithFlights> {
    return this.httpClient.post<CrewMembersWithFlights>(`${this.baseUrl}/create`, toCreate)
  }

  public updateCrewMember(toUpdate: CrewMember): Observable<CrewMembersWithFlights> {
    return this.httpClient.put<CrewMembersWithFlights>(`${this.baseUrl}/update`, toUpdate);
  }

  public delete(id: number): Observable<void> {
    return this.httpClient.delete<void>(`${this.baseUrl}/delete?id=${id}`);
  }
}
